_user/rhealstone/irq-latency/main.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Rhealstone benchmark
+ *
+ * Message latency
+ *
+ * Copyright 2024 Phoenix Systems
+ * Author: Lukasz Leczkowski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdatomic.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <sys/interrupt.h>
+#include <sys/mman.h>
+#include <sys/time.h>
+#include <sys/threads.h>
+
+#include <board_config.h>
+#include <rhcommon.h>
+
+
+#define BENCHMARKS 10000
+
+atomic_uint_least64_t benchEnd = 0;
+
+#ifdef __CPU_GR740
+#define IRQ_UNUSED 13
+#endif
+
+uint64_t results[BENCHMARKS] = { 0 };
+
+
+int irqHandler(unsigned int n, void *arg)
+{
+	if (n == IRQ_UNUSED) {
+		benchEnd = getCntr();
+	}
+
+	return 0;
+}
+
+
+int main(void)
+{
+	BENCH_NAME("Interrupt latency");
+
+	uint32_t *irqCtrl = mmap(NULL, _PAGE_SIZE, PROT_READ | PROT_WRITE, MAP_DEVICE | MAP_PHYSMEM | MAP_ANONYMOUS, -1, (uintptr_t)INT_CTRL_BASE);
```

Comment:
Should be volatile, actually it's kinda weird that the access to the `irqCtrl` hasn't been optimized out. Are we sure we have `-O2`?